tracer/src/Datadog.Trace/Debugger/SpanCodeOrigin/SpanCodeOriginManager.cs

```diff
@@ -5,94 +5,219 @@
 #nullable enable
 
 using System;
+using System.Collections.Concurrent;
 using System.Diagnostics;
+using System.Reflection;
 using Datadog.Trace.Debugger.Symbols;
 using Datadog.Trace.Logging;
+using Datadog.Trace.Pdb;
+using Datadog.Trace.Tagging;
 using Datadog.Trace.VendoredMicrosoftCode.System.Buffers;
+using Datadog.Trace.VendoredMicrosoftCode.System.Collections.Immutable;
 
 namespace Datadog.Trace.Debugger.SpanCodeOrigin
 {
     internal class SpanCodeOriginManager
     {
-        private const string CodeOriginTag = "_dd.code_origin";
+        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(SpanCodeOriginManager));
 
-        private const string FramesPrefix = "frames";
+        private readonly ConcurrentDictionary<Assembly, AssemblyPdbInfo?> _assemblyPdbCache = new();
 
-        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(SpanCodeOriginManager));
+        private readonly CodeOriginTags _tags;
+
+        private readonly int _maxUserFrames;
 
-        private readonly DebuggerSettings _settings = LiveDebugger.Instance?.Settings ?? DebuggerSettings.FromDefaultSource();
+        private readonly bool _isCodeOriginEnabled;
+
+        private readonly ImmutableHashSet<string> _thirdPartyDetectionExcludes;
+
+        private readonly ImmutableHashSet<string> _thirdPartyDetectionIncludes;
+
+        private SpanCodeOriginManager()
+        {
+            _tags = new CodeOriginTags(_maxUserFrames);
+            var settings = LiveDebugger.Instance?.Settings ?? DebuggerSettings.FromDefaultSource();
+            _maxUserFrames = settings.CodeOriginMaxUserFrames;
```

Comment:
You use `_maxUserFrames` before setting it

Reply:
Fixed in 5792875cc54444656eb03006177ac1fe4166d3f7